server/src/main/java/com/server/loan/calculator/services/CalculatorService.java

```diff
@@ -0,0 +1,60 @@
+package com.server.loan.calculator.services;
+
+import com.server.loan.calculator.exception.ValueOutOfRangeException;
+import com.server.loan.calculator.models.CalculatorData;
+import com.server.loan.calculator.validators.InputAsCorrectNumberValidator;
+import com.server.loan.calculator.validators.ValueInRangeValidator;
+import lombok.Data;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+@Service
+@Data
+public class CalculatorService {
+
+    private final CalculatorData data;
+    private final ValueInRangeValidator valueInRangeValidator;
+    private final InputAsCorrectNumberValidator inputValidator;
+
+    private double calculateMonthlyInterestRateDecimals () {
+        return ((data.getEuriborInterestRate() + data.getBankInterestMargin()) / 100) / 12;
+    }
+
+    private BigDecimal calculateMaxAvailableLoan () {
+        inputValidator.validate(data.getSalary());
+        inputValidator.validate(data.getFinancialObligations());
+        inputValidator.validate(data.getMortgagePeriodYears());
+        valueInRangeValidator.validate(data.getMortgagePeriodYears(), 1, 30);
+        double monthlyAvailableMoney = (data.getSalary() * 0.4) - (data.getFinancialObligations());
+        BigDecimal mathPower = BigDecimal.valueOf(Math.pow(1 + (calculateMonthlyInterestRateDecimals()), -(12 * data.getMortgagePeriodYears())));
+        return (BigDecimal.valueOf(monthlyAvailableMoney)
+                .multiply((BigDecimal.ONE.subtract(mathPower)))).divide(BigDecimal.valueOf(calculateMonthlyInterestRateDecimals()), RoundingMode.HALF_UP);
+    }
+
+    private BigDecimal calculateMonthlyPayment () {
+        valueInRangeValidator.validate(data.getPropertyValue(), data.getMinPropertyValue(), data.getMaxPropertyValue());
```

Comment:
all this validation should be moved to a separate class, and should be called before calling calculation methods